numpy/f2py/func2subr.py

```diff
@@ -130,7 +130,7 @@ def add(line, ret=ret):
             l = l + ', ' + fortranname
     if need_interface:
         for line in rout['saved_interface'].split('\n'):
-            if line.lstrip().startswith('use '):
+            if line.lstrip().startswith('use ') and not '__user__' in line:
```

Comment:
Is `__user__` an existing f2py thing, or something new you're adding here?

Reply:
This is an f2py thing and also checked in other places as an exceptional case.

Reply:
... even more explicitly: This is the intended and documented behaviour of f2py that also occurs if you don't trigger the bug, i.e. don't define an assumed shape array in the same subroutine.

Reply:
@eric-wieser what are the further steps now? As far as I can tell the PR fixes the bug and introduces only minimal changes to reproduce documented behaviour.